store/snapshots/store.go

```diff
@@ -170,16 +171,21 @@ func (s *Store) Load(height uint64, format uint32) (*types.Snapshot, <-chan io.R
 			chunk, err := s.loadChunkFile(height, format, i)
 			if err != nil {
 				_ = pw.CloseWithError(err)
+				_ = chunk.Close()
```

Comment:
This isn't correct and will cause a nil pointer panic. chunk most likely will be nil by the time an error is returned given that the idiom in Go is to the either return (nil object, non-nil error) or return (non-nil object, nil error)